source/client_channel_handler.c

```diff
@@ -24,6 +24,18 @@
  * Packet State Machine
  ******************************************************************************/
 
+static void s_mqtt_connection_lock_synced_data(struct aws_mqtt_client_connection *connection) {
```

Comment:
Maybe instead of having 2 copies of these functions, just put them in one place without static (probably client.c) and extern them to the other.